PortalSwift/Sources/PortalSwift/Storage/GDrive/GDriveClient.swift

```diff
@@ -0,0 +1,305 @@
+//
+//  GDriveClient.swift
+//  PortalSwift
+//
+//  Created by Blake Williams on 2/7/23.
+//
+
+import Foundation
+import GoogleSignIn
+
+public enum GDriveClientError: Error {
+  case authenticationNotInitialized(String)
+  case fileContentMismatch
+  case noFileFound
+  case unableToBuildGDriveQuery
+  case unableToDeleteFromGDrive
+  case unableToReadFileContents
+  case unableToWriteToGDrive
+  case userNotAuthenticated
+  case viewNotInitialized(String)
+}
+
+public class GDriveClient {
+  public var auth: GoogleAuth?
+  public var clientId: String? {
+    get { return self._clientId }
+    set(clientId) {
+      self._clientId = clientId
+
+      if let clientId = clientId, let view = view {
+        self.auth = GoogleAuth(config: GIDConfiguration(clientID: clientId), view: view)
+      }
+    }
+  }
+
+  public var folder: String
+  public var view: UIViewController? {
+    get { return self._view }
+    set(view) {
+      self._view = view
+
+      if let clientId = clientId, let view = view {
+        self.auth = GoogleAuth(config: GIDConfiguration(clientID: clientId), view: view)
+      }
+    }
+  }
+
+  private var _clientId: String?
+  private var _view: UIViewController?
+  private var api: HttpRequester
```

Comment:
is api needed in this class anymore. It doesnt look like its used

Reply:
It's used to get the clientId. The API is now responsible for housing the latest `/clients/me` response returned.